frontend/src/metabase/query_builder/components/saved-question-picker/SavedQuestionList.jsx

```diff
@@ -43,46 +44,50 @@ function SavedQuestionList({
     </SavedQuestionListEmptyState>
   );
 
-  const isVirtualCollection = collection.id === PERSONAL_COLLECTIONS.id;
+  const isVirtualCollection = collection?.id === PERSONAL_COLLECTIONS.id;
   const schemaId = getCollectionVirtualSchemaId(collection, { isDatasets });
 
   return (
     <SavedQuestionListRoot>
-      {!isVirtualCollection && (
-        <Schemas.Loader id={schemaId}>
-          {({ schema }) => {
-            const tables =
-              databaseId != null
-                ? schema.tables.filter(table => table.db_id === databaseId)
-                : schema.tables;
-            return (
-              <React.Fragment>
-                {tables
-                  .sort((a, b) => a.display_name.localeCompare(b.display_name))
-                  .map(t => (
-                    <SavedQuestionListItem
-                      id={t.id}
-                      isSelected={selectedId === t.id}
-                      key={t.id}
-                      size="small"
-                      name={t.display_name}
-                      icon={{
-                        name: isDatasets ? "model" : "table2",
-                        size: 16,
-                      }}
-                      onSelect={() => onSelect(t)}
-                      rightIcon={PLUGIN_MODERATION.getStatusIcon(
-                        t.moderated_status,
-                      )}
-                    />
-                  ))}
-                {tables.length === 0 ? emptyState : null}
-              </React.Fragment>
-            );
-          }}
-        </Schemas.Loader>
-      )}
-      {isVirtualCollection && emptyState}
+      <LoadingWrapper loading={!collection}>
+        {!isVirtualCollection && (
+          <Schemas.Loader id={schemaId}>
+            {({ schema }) => {
+              const tables =
+                databaseId != null
+                  ? schema.tables.filter(table => table.db_id === databaseId)
+                  : schema.tables;
+              return (
+                <React.Fragment>
+                  {tables
+                    .sort((a, b) =>
+                      a.display_name.localeCompare(b.display_name),
+                    )
+                    .map(t => (
```

Comment:
I see we didn't really want to refactor anything here, we simply put this block inside a `LoadingWrapper`.

At the same time, a couple ideas that could make this block more palatable.

1. Instead of using `t` as the map's iterated item name, to use `table`. 
2. Extracting what's in this map into a function and reducing nesting depth.